lib/hocon/impl/path.rb

```diff
@@ -35,11 +26,85 @@ def self.has_funky_chars?(s)
   end
 
   def initialize(first, remainder)
+    # first: String, remainder: Path
+
     @first = first
     @remainder = remainder
   end
   attr_reader :first, :remainder
 
+  def self.from_string_list(elements)
+    # This method was translated from the Path constructor in the
+    # Java hocon library that has this signature:
+    #   Path(String... elements)
+    #
+    # It figures out what @first and @remainder should be, then
+    # pass those to the ruby constructor
+    if elements.length == 0
+      raise Hocon::ConfigError::ConfigBugOrBrokenError("empty path")
+    end
+
+    new_first = elements.first
+
+    if elements.length > 1
+      pb = Hocon::Impl::PathBuilder.new
+
+      # Skip first element
+      elements.drop(1).each do |element|
+        pb.append_key(element)
+      end
+
+      new_remainder = pb.result
+    else
+      new_remainder = nil
+    end
+
+    self.new(new_first, new_remainder)
+  end
+
+  def self.from_path_list(path_list)
+    # This method was translated from the Path constructors in the
+    # Java hocon library that take in a list of Paths
+    #
+    # It just passes an iterator to self.from_path_iterator, which
+    # will return a new Path object
+    from_path_iterator(path_list.each)
+  end
+
+  def self.from_path_iterator(path_iterator)
+    # This method was translated from the Path constructors in the
+    # Java hocon library that takes in an iterator of Paths
+    #
+    # It figures out what @first and @remainder should be, then
+    # pass those to the ruby constructor
+
+    # Try to get first path from iterator
+    # Ruby iterators have no .hasNext() method like java
+    # So we try to catch the StopIteration exception
```

Comment:
This comment seems a bit off ... it seems like the actual issue is that, in 1.9.3, there's no `size` method on the `Enumerator` class.  Is that right?